compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.cpp

```diff
@@ -20,51 +21,104 @@ using llvm::APIntOps::GreatestCommonDivisor;
 
 namespace mlir::iree_compiler {
 
+template <typename T>
+static llvm::raw_ostream &operator<<(llvm::raw_ostream &os,
+                                     const llvm::SmallVectorImpl<T> &vector) {
+  os << "[";
+  llvm::interleaveComma(vector, os);
+  os << "]";
+  return os;
+}
+
 static llvm::raw_ostream &operator<<(llvm::raw_ostream &os,
```

Comment:
Slightly unrelated, but would be good to make this operator publicly exposed. It's nice to be able to print this for debugging.